pkg/controller/v1alpha2/applicationconfiguration/render.go

```diff
@@ -659,3 +674,38 @@ func getTraitName(ac *v1alpha2.ApplicationConfiguration, componentName string,
 
 	return traitName
 }
+
+// getExistingWorkload tries to retrieve the currently running workload
+func (r *components) getExistingWorkload(ctx context.Context, ac *v1alpha2.ApplicationConfiguration, c *v1alpha2.Component, w *unstructured.Unstructured) *unstructured.Unstructured {
+	var workloadName string
+	existingWorkload := &unstructured.Unstructured{}
+	for _, component := range ac.Status.Workloads {
+		if component.ComponentName != c.GetName() {
+			continue
+		}
+		workloadName = component.Reference.Name
```

Comment:
Maybe adding `exit` after this line is better?

Reply:
break